src/app/screens/SignTransaction/SignTransactionMnemonic.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+
+import { getAddress } from '@app/services';
+import { GetMnemonicAddressesResult, SignTransactionProps, WalletType } from '@types';
+
+export const SignTransactionMnemonic = ({
+  onAccept,
+  onDeny,
+  hasPersistentPrivateKey,
+  currentAccount,
+  tx
+}: SignTransactionProps) => {
+  const [phrase, setPhrase] = useState('');
+  const [password, setPassword] = useState('');
+
+  const changeMnemonicPhrase = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setPhrase(e.currentTarget.value);
+
+  const changePassword = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setPassword(e.currentTarget.value);
+
+  const handleAccept = async () => {
+    if (!hasPersistentPrivateKey) {
+      const { privateKey } = (await getAddress({
```

Comment:
This will throw an uncaught error if the mnemonic is invalid. Form input should be checked, and promise rejections should be handled.